docs-md/concepts/processing-guarantees.md

```diff
@@ -1,11 +1,93 @@
 ---
 layout: page
 title: Processing Guarantees
-tagline: Processing in ksqlDB
-description: Learn about processing in ksqlDB
-keywords: ksqldb, processing
+tagline: Processing guarantees in ksqlDB
+description: Learn about at-least-once, exactly-once semantics in ksqlDB
+keywords: ksqldb, processing, eos, at-least-once, exactly-once
 ---
 
-TODO: EOS, etc.
+ksqlDB supports *at-least-once* and *exactly-once* processing guarantees.
 
-Page last revised on: {{ git_revision_date }}
\ No newline at end of file
+At-least-once semantics
+-----------------------
+
+Records are never lost but may be redelivered. If your stream processing
+application fails, no data records are lost and fail to be processed, but some
+data records may be re-read and therefore re-processed. At-least-once semantics
+is enabled by default in your ksqlDB configuration, with
+`processing.guarantee="at_least_once"`.
+
+Exactly-once semantics
+----------------------
+
+Records are processed once. Even if a producer sends a duplicate record, it's
+written to the broker exactly once.
+
+Exactly-once stream processing is the ability to execute a read-process-write
+operation exactly one time. All of the processing happens exactly once,
+including the processing and the materialized state created by the processing
+job that is written back to {{ site.ak }}.
+
+To enable exactly-once semantics, set `processing.guarantee="exactly_once"` in
+your ksqlDB configuration.
+
+For more information, see
+[Processing Guarantees](https://docs.confluent.io/current/streams/concepts.html#processing-guarantees).
+
+### Enable exactly-once semantics
+
+Exactly-once isn't enabled by default in ksqlDB, but you can enable it on a
+query-by-query basis by passing the `processing.guarantee` configuration setting
+to ksqlDB.
+
+How you pass the configuration setting to ksqlDB depends on how you
+run ksqlDB Server and on how you send requests to start queries.
+
+#### ksqlDB CLI
+
+Use the SET command to enable exactly-once for the subsequent
+query:
+
+```sql
+SET 'processing.guarantee' = 'exactly_once';
+```
+
+For more information, see
+[Configure ksqlDB CLI](../operate-and-deploy/installation/cli-config.md).
+
+#### REST API
+
+Pass the config as a property along with the request:
+
+```http
+POST /query HTTP/1.1
+Accept: application/vnd.ksql.v1+json
+Content-Type: application/vnd.ksql.v1+json
+
+{
+"ksql": "SELECT * FROM pageviews EMIT CHANGES;",
+"streamsProperties": {
+    "processing.guarantee": "exactly_once"
+}
+}
+```
+
+For more information, see
+[Run a query and stream back the output](../developer-guide/ksqldb-rest-api/query-endpoint.md).
+
+#### Default for all queries and non-interactive (headless) mode
+
+To enable exactly-once by default for all queries, and for non-interactive
+(headless) mode, set the configuration in the ksqlDB Server properties file,
+which by default is located at
+`${CONFLUENT_HOME}/etc/ksql/ksql-server.properties` in a {{ site.cp }}

```

Comment:
Was there an effort to remove references to CP from the community docs?

Reply:
@vcrfxia yes, and I stuck this in here just because there's no convenient place in the Enterprise docs (yet).

Reply:
I added an example for Docker, too.